CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+contributors to follow so that we can keep the codebase consistent and clean.
+
+* Fork the main repository and clone it to your local machine.
+* Create a topic branch from master. Please avoid working directly on the ```master``` branch.
+* Make commits of logical units, and make sure to follow the CODINGSTYLE.
+* Submit the Pull Request(PR) on Github.
+
+## Issues
+We use GitHub [issues tracker](https://github.com/Scribery/tlog/issues) to track bugs.
+Please ensure your description is clear and has sufficient instructions; so we are able to reproduce the issue.
+
+## Coding Style guidelines
+
+* Use four spaces for indentation, please don't use tabs except where required, such as in Makefiles.
+
+* Line should be at most 78 chars.
+
+* Please don't leave trailing whitespace at the ends of lines.
+
+* Braces open on the same line as the `for`/`while`/`if`/`else`/`switch`/etc. Closing
+  braces are put on a line of their own, except in the `else` of an `if` statement
+  or in a while of a `do`-`while` statement. Always use braces for `if` and `while`.
+
+```c
+if (a == b) {
+    ...
+}
+
+if (a == b) {
+    ...
+} else if (a > b) {
+    ...
+}
+
+if (a == b) {
+    do_something();
+} else {
+    do_something_else();
+}
+
+do {
+    do_something();
+} while (cond);
+
+while (cond) {
+    do_something();   
+}
+
+switch (expression) {
+case 'constant-expression':
+    ...
+case 'constant-expression':
+    ...
+default:
+    ...
+}
+
+```
+
+* In general, do not use `goto`. The goto statement only comes in handy when a
+  function exits from multiple locations and some common work such as cleanup
+  has to be done. If there is no cleanup needed then just return directly.
+
+* For function, `static` and return type are put on the same line. Function's name is put on a
+  line of its own.
+
+```c
+static tlog_grc
+tlog_play_run(struct tlog_errs **perrs, int *psignal)
+{
+              ...
+}
+```
+
+* For regular comments, use C style (`/*` `*/`) comments, even for one-line.
+
+* For doc-comments (Doxygen comments), use the `/**` `*/` style featured.
+
+* For wrapping (`\`) of Macro, make sure to follow:
+
+```c
+#define CMP_BOOL(_name) \
+    do {                                                        \
+        if (_name != t._name##_out) {                           \
+            FAIL(#_name " %s != %s",                            \
+                 BOOL_STR(_name), BOOL_STR(t._name##_out));     \
+        }                                                       \
+    } while (0)
+```
+
+## Directories
+Tlog source is organized as follows.
+
+```
+.                                           <- Top-level build and documentation files
+├── doc                                     <- Documentation.
+├── include                                 <- C headers.
+│   ├── tlog
+│   ├── tltest
+│   │   ├── json_sink.h
+│   │   ├── json_source.h
+│   │   ├── json_stream_enc.h
+│   │   ├── misc.h
```

Comment:
Don't put the files here. Just describe what `tlog` and `tltest` directories contain. E.g.

    ├── include         <- C headers
    │   ├── tlog        <- Tlog library headers
    │   ├── tltest      <- Tlog test library headers

And so on for all the second-level directories.

Also, please don't put full stops at the end of those descriptions, they
create too much noise and make this harder to read.